plugins/azure-devops-backend/src/utils/azure-devops-utils.ts

```diff
@@ -263,3 +289,36 @@ function convertCreatedBy(identityRef?: IdentityRef): CreatedBy | undefined {
 function hasAutoComplete(pullRequest: GitPullRequest): boolean {
   return pullRequest.isDraft !== true && !!pullRequest.completionOptions;
 }
+
+export function extractAssets(content: string) {
+  const regExp =
+    /\[([^\[\]]*)\]\((?!https?:\/\/)(.*?)(\.png|\.jpg|\.jpeg|\.gif|\.webp)(.*)\)/gim;
+  return content.match(regExp) || [];
+}
+
+export function extractPartsFromAsset(content: string): {
+  label: string;
+  path: string;
+  ext: string;
+} {
+  const regExp =
+    /\[(.*?)\]\((?!https?:\/\/)(.*?)(\.png|\.jpg|\.jpeg|\.gif|\.webp)(.*)\)/;
+  const [_, label, path, ext] = regExp.exec(content) || [];
+  return {
+    ext,
+    label,
+    path: path.startsWith('.') ? path.substring(1, path.length) : path,
+  };
+}
+
+export function getMimeByExtension(ext: string): string {
```

Comment:
i see we use the `mime-types` library in one other place at least, would that help here? https://github.com/backstage/backstage/blob/87649a06bfc38d39646117f4e5f80cd3ddf9d724/plugins/techdocs-node/package.json#L60